lib/src/client/socks_client.dart

```diff
@@ -194,8 +194,7 @@ class SocksSocket with StreamMixin<Uint8List>, SocketMixin, ByteReader {
         0x00, // Reserved
         addressType.byte,
         // Encoding address, if domain adding length at the beginning.
-        if (addressType == AddressType.domain) rawAddress.length,
-        ...rawAddress,
+        ...(addressType == AddressType.domain ? [rawAddress.length - 1, ...rawAddress.sublist(0, rawAddress.length - 1)] : rawAddress),
```

Comment:
Does domain ALWAYS have null terminator, or we're just cutting any last byte? Shouldn't there be a check for null terminator or better a valid handling of it.